python/beeai_framework/agents/tool_calling/agent.py

```diff
@@ -93,7 +93,10 @@ async def run(self, input: str | list[AnyMessage], /, **kwargs: Unpack[AgentOpti
         Returns:
             The agent output.
         """
+        if not input:
+            raise ValueError("Invalid input. The input must be a non-empty string or list of messages.")
```

Comment:
Also, breaking change. Empty input is valid.

Reply:
Do we want to still accept empty inputs in the agent implementations? If so, what should the treatment be? Maybe, something like this?

```Python
if not input and memory.is_empty():
    raise ValueError("Invalid input. The input must be a non-empty string or list of messages.")
```

Reply:
I added the check above. That should be a good validation to have.